docs/proposals/proposal-audit-log.md

```diff
@@ -0,0 +1,265 @@
+# Audit log backend part
+
+### Purpose of audit log
+
+The audit log is a feature that allows users to track changes made to their organization, projects,
+schemas, and other resources.
+
+
+## Implementation
+
+Proposed implementation is to use a single Clickhouse table to store all events. The table will have
+the following schema:
+
+
+```sql
+CREATE TABLE audit_log (
+  timestamp DateTime('UTC') CODEC(DoubleDelta, LZ4),
+  user_id LowCardinality(String) CODEC(ZSTD(1)),
+  user_email STRING,
+  organization_id LowCardinality(String) CODEC(ZSTD(1)),
+  project_id LowCardinality(String) CODEC(ZSTD(1)),
+  project_name STRING,
+  target_id LowCardinality(String) CODEC(ZSTD(1)),
+  target_name STRING,
+  schema_version_id LowCardinality(String) CODEC(ZSTD(1)),
+  event_action LowCardinality(String) CODEC(ZSTD(1)),
+  event_details JSON,
```

Comment:
```
Set setting allow_experimental_object_type = 1 in order to allow it. (ILLEGAL_COLUMN) (version 24.4.3.25 (official build))"
}
```

actually if we want to use JSON, we would have to enable `allow_experimental_object_type`.
Do you want to enable that or change this to plain string @kamilkisiela ?